mz_os_posix.c

```diff
@@ -46,28 +46,25 @@
 #if defined(HAVE_ICONV)
 char *mz_os_utf8_string_create(const char *string, int32_t encoding) {
     iconv_t cd;
+    // up to CP2147483647
+    char string_encoding[13];
     const char *from_encoding = NULL;
     size_t result = 0;
     size_t string_length = 0;
     size_t string_utf8_size = 0;
     char *string_utf8 = NULL;
     char *string_utf8_ptr = NULL;
 
-    if (!string)
+    if (!string || encoding <= 0)
         return NULL;
 
-    if (encoding == MZ_ENCODING_CODEPAGE_437)
-        from_encoding = "CP437";
-    else if (encoding == MZ_ENCODING_CODEPAGE_932)
-        from_encoding = "CP932";
-    else if (encoding == MZ_ENCODING_CODEPAGE_936)
-        from_encoding = "CP936";
-    else if (encoding == MZ_ENCODING_CODEPAGE_950)
-        from_encoding = "CP950";
-    else if (encoding == MZ_ENCODING_UTF8)
+    // actual list of supported encodings: `iconv --list`
```

Comment:
Remove comment, since iconv is not on Windows.

Reply:
The file is `mz_os_posix.c`, so it's not really for Windows.